Controllers/CommandsController.cs

```diff
@@ -0,0 +1,263 @@
+    // If no channel is specified, use the current channel
+    channel ??= (IGuildChannel)cmd.Channel;
+
+    // If eatExisting is true but channel isn't a forum, respond with an error message
+    if (eatExisting && channel.GetChannelType() != ChannelType.Forum) {
+      cmd.RespondAsync(Locale.Get("output-eatexisting-notforum"), null, false, true);
+      return false;
+    }
+
+    // If eatFuture is true but channel isn't a forum, respond with an error message
+    if (eatFuture && channel.GetChannelType() != ChannelType.Forum) {
+      cmd.RespondAsync(Locale.Get("output-eatfuture-notforum"), null, false, true);
+      return false;
+    }
+
+    // If both eatExisting and eatFuture are disabled on forum
+    if (channel.GetChannelType() == ChannelType.Forum && !eatExisting && !eatFuture) {
+      cmd.RespondAsync(Locale.Get("output-whatdoido"), null, false, true);
+      return false;
+    }
+
+    return true;
+  }
+
+  private static async void Eat(IGuildChannel channel, long timeType, long timeValue, bool eatExisting, bool eatFuture, SocketSlashCommand cmd) {
+    TimeSpan eatIn = TimeSpan.FromSeconds(timeValue * timeType);
+    DateTime eatTime = DateTime.UtcNow + eatIn;
+
+    if (channel.GetChannelType() == ChannelType.PublicThread) { // If thread
+      (ThreadService.Result, DateTime?) res = await ThreadService.EatThread(channel, eatTime);
+      EatThreadResponse(channel, eatIn, res, cmd);
+    } else { // If Forum
+      (ForumService.Result, TimeSpan?) res = await ForumService.EatForum(channel, eatIn, eatExisting, eatFuture);
+      EatForumResponse(channel, eatIn, eatExisting, eatFuture, res, cmd);
+    }
+  }
+
+  private static async void EatThreadResponse(IGuildChannel channel, TimeSpan eatIn, (ThreadService.Result, DateTime?) res, SocketSlashCommand cmd) {
+    switch (res.Item1) {
+      case ThreadService.Result.Success:
+        await cmd.RespondAsync(Locale.Get("output-eat-thread-success", channel.Name, eatIn.ToDynamicString()), null, false, true);
+        break;
+      case ThreadService.Result.WrongChannelType:
+        await cmd.RespondAsync(Locale.Get("output-eat-wrongchanneltype"), null, false, true);
+        break;
+      case ThreadService.Result.Overwrote:
+        TimeSpan overwroteETA = (DateTime)res.Item2! - DateTime.UtcNow;
+        await cmd.RespondAsync(Locale.Get("output-eat-thread-overwrote", overwroteETA.ToDynamicString(), channel.Name, eatIn.ToDynamicString()), null, false, true);
+        break;
+      case ThreadService.Result.ForumOverride:
+        TimeSpan forumOverrideETA = (DateTime)res.Item2! - DateTime.UtcNow;
+        await cmd.RespondAsync(Locale.Get("output-eat-thread-forumoverride", forumOverrideETA.ToDynamicString(), channel.Name, eatIn.ToDynamicString()), null, false, true);
+        break;
+    }
+  }
+
+  private static async void EatForumResponse(IGuildChannel channel, TimeSpan eatIn, bool eatExisting, bool eatFuture, (ForumService.Result, TimeSpan?) res, SocketSlashCommand cmd) {
```

Comment:
You can name properties in a Tuple. 

```suggestion
  private static async void EatForumResponse(IGuildChannel channel, TimeSpan eatIn, bool eatExisting, bool eatFuture, (ForumService.Result result, TimeSpan? eta) res, SocketSlashCommand cmd) {
```

So you don't have to live with `Item1` and `Item2`)

Reply:
Fixed with 312fbd80d615684e398d53bc1d6aeae7e01922ec